api_docs/trashbin.apib

```diff
@@ -0,0 +1,344 @@
+            "audit": { }
+        }
+
+## Restore Trashbin Item [/trashbin/{object_id}/{object_kind}/restore]
++ Parameters
+    + object_kind (string, required, `dds-file`) ... The kind of object.
+    + object_id (string, required, `777be35a-98e0-4c2e-9a17-7bc009f9b111`) ... The unique id of the object.
+
+### Restore Trashbin Item instance [PUT]
+>**Permission:** create_file or system_admin
+- - - - -
+
+>**Rules**
+- - - - -
++ If parent is not supplied in the payload, an attempt will be made to restore to the original parent.
++ An exception is returned if the parent is itself in the trash bin. You must restore the parent before you can restore a child to it.
++ `dds-file` and `dds-folder` objects can only be restored to a `dds-project` or `dds-folder` parent.
++ If a `dds-file` is restored, all deleted version history will be restored as well.
++ `dds-file-version` objects can only be restored to their original "owning" file.
++ If a `dds-file-version` is restored, all deleted version history of the original "owning" file will be restored as well.
+
+**Response messages:**
+* 201: Valid API Token in 'Authorization' Header
+* 400: Parent object does not exist, has been purged, or is in the trash bin
+* 401: Missing, Expired, or Invalid API Token in 'Authorization' Header
+* 403: Forbidden
+* 404: Item does not exist in the trash bin
+
++ Response 201 (application/json)
+
+        {
+            "kind": "dds-folder",
+            "id": "d5ae02a4-b9e6-473d-87c4-66f4c881ae7a",
+            "parent": {
+                "kind": "dds-folder",
+                "id": "552be8c5-209d-4e3b-afaf-cb66686ffbyy"
+            },
+            "name": "Raw Data",
+            "project": {
+                "id": "ca29f7df-33ca-46dd-a015-92c46fdb6fd1"
+            },
+            "ancestors": [
+                {
+                    "kind": "dds-project",
+                    "id": "ca29f7df-33ca-46dd-a015-92c46fdb6fd1",
+                    "name": "Knockout Mouse Project (KOMP)"
+                },
+                {
+                    "kind": "dds-folder",
+                    "id": "552be8c5-209d-4e3b-afaf-cb66686ffbyy",
+                    "name": "Sequencing Core"
+                }
+            ],
+            "is_deleted": false,
+            "audit": { }
+        }
+
+## Purge Trashbin Item [/trashbin/{object_id}/{object_kind}/purge]
++ Parameters
+    + object_kind (string, required, `dds-file`) ... The kind of object.
+    + object_id (string, required, `777be35a-98e0-4c2e-9a17-7bc009f9b111`) ... The unique id of the object.
+
+### Purge Trashbin Item instance [PUT]
+>**Permission:** create_file or system_admin
+- - - - -
+**Response messages:**
+* 201: Valid API Token in 'Authorization' Header
+* 400: Parent object does not exist, has been purged, or is in the trash bin
+* 401: Missing, Expired, or Invalid API Token in 'Authorization' Header
+* 403: Forbidden
+* 404: Item does not exist in the trash bin
+
++ Response 201
+
```

Comment:
Wait... Where's the rest of the response? If this doesn't return a body, like a `DELETE` call, then it should probably return a `204`.